thor/src/protocols/splice.rs

```diff
@@ -99,20 +157,52 @@ impl State0 {
         })
         .expect("Only fails if script_sig or witness is empty which is not the case.");
 
+        let mut our_balance = self.previous_balance.ours;
+        let mut their_balance = self.previous_balance.theirs;
+
+        let mut splice_in_inputs = vec![];
+
+        if let Some(splice_in) = self.splice_in_self.clone() {
+            splice_in_inputs.push(splice_in.input_psbt);
+            our_balance += splice_in.amount;
+        }
+
+        if let Some(splice_in) = splice_in_other {
+            splice_in_inputs.push(splice_in.input_psbt);
+            their_balance += splice_in.amount;
+        }
+
+        // Sort the PSBT inputs based on the ascending lexicographical order of
+        // bytes of their consensus serialization. Both parties _must_ do this so that
+        // they compute the same splice transaction.
+        splice_in_inputs.sort_by(|a, b| {
+            serialize(a)
+                .partial_cmp(&serialize(b))
+                .expect("comparison is possible")
+        });
+
+        // The previous funding psbt MUST be the first input
```

Comment:
Because then we don't know to which input (previous TX_f output) should we add the (X_0, X_1) signatures.
But now I realise that we could look for the input using the script pubkey... 🤔

Reply:
> But now I realise that we could look for the input using the script pubkey

I think that would be better. Both parties should always be able to compute the shared output. And it might make it easier for you to do the sorting inside if you don't need to enforce an order on the outside.